dotCMS/src/main/java/com/dotmarketing/util/ImportStarterUtil.java

```diff
@@ -0,0 +1,1148 @@
+            APILocator.getIdentifierAPI().save(ident);
+        }
+
+
+
+        // we store here defaultFileType for every folder
+        // because of mutual folder <--> structure dependency
+        final Map<String, String> fileTypesInodes = new HashMap<String, String>();
+
+        // now we can import folders
+        for (File ff : contains(Folder.class.getCanonicalName())) {
+            try {
+                doXMLFileImport(ff, new ObjectFilter() {
+                    public boolean includeIt(Object obj) {
+                        Folder f = (Folder) obj;
+                        fileTypesInodes.put(f.getInode(), f.getDefaultFileType());
+                        f.setDefaultFileType(null);
+                        return true;
+                    }
+                });
+            } catch (Exception e) {
+                Logger.error(this, "Unable to load " + ff.getName() + " : " + e.getMessage(), e);
+            }
+        }
+
+
+        // we need content types before contentlets
+        // but content types have references to folders and hosts identifiers
+        // so, here is the place to do it
+        for (File file : endsWith(ContentTypeImportExportUtil.CONTENT_TYPE_FILE_EXTENSION)) {
+            new ContentTypeImportExportUtil().importContentTypes(file);
+        }
+
+        // updating file_type on folder now that structures were added
+        DotConnect dc = new DotConnect();
+        for (Entry<String, String> entry : fileTypesInodes.entrySet()) {
+            dc.setSQL("update folder set default_file_type=? where inode=?");
+            dc.addParam(entry.getValue());
+            dc.addParam(entry.getKey());
+            dc.loadResult();
+        }
+
+
+        for (File file : contains(Relationship.class.getCanonicalName() + "_")) {
+            doXMLFileImport(file);
+        }
+
+        // We have all identifiers, structures and users. Ready to import contentlets!
+        for (File file : contains(Contentlet.class.getCanonicalName() + "_")) {
+            doXMLFileImport(file);
+        }
+
+
+        for (File file : contains(Template.class.getCanonicalName() + "_")) {
+            doXMLFileImport(file);
+        }
+
+        for (File file : contains("com.dotmarketing.beans.TemplateContainers_")) {
+            doXMLFileImport(file);
+        }
+        for (File file : contains(Link.class.getCanonicalName() + "_")) {
+
+            doXMLFileImport(file);
+        }
+        for (File file : contains(Container.class.getCanonicalName()+ "_")) {
+            doXMLFileImport(file);
+        }
+        
+        for (File file : contains("com.dotmarketing.beans.ContainerStructure_")) {
+            doXMLFileImport(file);
+        }
+        
+        for (File file : contains("com.dotmarketing.beans.Tree_")) {
+            doXMLFileImport(file);
+        }
+
+        // workflow schemas need to come before permissions
+        for (File file : contains("WorkflowSchemeImportExportObject.json")) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'file' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)